cpp/ql/test/library-tests/dataflow/taint-tests/stl.cpp

```diff
@@ -237,19 +237,19 @@ void sink(char) {}
 void test_range_based_for_loop() {
 	std::string s(source());
 	for(char c : s) {
-		sink(c); // tainted [NOT DETECTED]
+		sink(c); // tainted [NOT DETECTED by IR]
 	}
 
 	for(std::string::iterator it = s.begin(); it != s.end(); ++it) {
 		sink(*it); // tainted [NOT DETECTED]
 	}
 
 	for(char& c : s) {
-		sink(c); // tainted [NOT DETECTED]
+		sink(c); // tainted [NOT DETECTED by IR]
 	}
 
 	const std::string const_s(source());
 	for(const char& c : const_s) {
-		sink(c); // tainted [NOT DETECTED]
+		sink(c); // tainted [NOT DETECTED by IR]
```

Comment:
Is it possible to add a test case that doesn't involve strings?

Reply:
Yes. I've now added a test using `std::vector`.

Reply:
That's fantastic, thanks!